lib/Sema/DerivedConformanceEquatableHashable.cpp

```diff
@@ -1045,29 +1029,30 @@ deriveBodyHashable_hashValue(AbstractFunctionDecl *hashValueDecl) {
 }
 
 /// Derive a 'hashValue' implementation.
-static ValueDecl *
-deriveHashable_hashValue(TypeChecker &tc, Decl *parentDecl,
-                         NominalTypeDecl *typeDecl) {
+static ValueDecl *deriveHashable_hashValue(DerivedConformance &derived) {
   // @derived var hashValue: Int {
   //   return _hashValue(for: self)
   // }
+  auto &tc = derived.TC;
   ASTContext &C = tc.Context;
 
-  auto parentDC = cast<DeclContext>(parentDecl);
+  auto parentDC = derived.getConformanceContext();
   Type intType = C.getIntDecl()->getDeclaredType();
 
   // We can't form a Hashable conformance if Int isn't Hashable or
   // ExpressibleByIntegerLiteral.
-  if (!tc.conformsToProtocol(intType,C.getProtocol(KnownProtocolKind::Hashable),
-                             typeDecl, None)) {
-    tc.diagnose(typeDecl->getLoc(), diag::broken_int_hashable_conformance);
+  if (!tc.conformsToProtocol(intType,
+                             C.getProtocol(KnownProtocolKind::Hashable),
+                             derived.Nominal, None)) {
+    tc.diagnose(derived.Nominal->getLoc(),
+                diag::broken_int_hashable_conformance);
     return nullptr;
   }
 
   ProtocolDecl *intLiteralProto =
       C.getProtocol(KnownProtocolKind::ExpressibleByIntegerLiteral);
-  if (!tc.conformsToProtocol(intType, intLiteralProto, typeDecl, None)) {
-    tc.diagnose(typeDecl->getLoc(),
+  if (!tc.conformsToProtocol(intType, intLiteralProto, derived.Nominal, None)) {
+    tc.diagnose(derived.Nominal->getLoc(),
```

Comment:
This diagnostic probably never fires, but `derived.ConformanceDecl` would be more appropriate for these two uses.